### PR TITLE
Moving to a folder structure which is more accessible

### PR DESCRIPTION
**Pull request by IshaanDesai**
This PR changes the folder structure of the entire repository to a more accessible form. This new folder structure has been decided upon based on inputs from students.

### CODE REVIEW

**Review by uekerman — CHANGES_REQUESTED** [review]
Goes in the right direction. A few things still:

* Please update the information in the main readme. We could also list the chapters here.
* The links in `timetable.md` still need to be updated.
* `git_basics_demo.md` needs further splitting. At least "My favorite neat little Git trick" should be separate.
* `containers_demo.md` needs further splitting.
* `homework_slides.md` -> `containers_homework_slides.md`
* `vm_containers_slides.md` -> `vm_containers_exercise`.md in chapter README
* `packaging_debian_slides.md` and similar: remove "packaging"
* hpc = spack

**Comment by ajaust** [comment]
Just as a note that the `hpc` pacakges should have `spack` in their name in `main` already. This could maybe fixed by a rebase.

**Comment by IshaanDesai** [comment]
First checklist:

- [ ] Please update the information in the main readme. We could also list the chapters here.
- [ ] The links in `timetable.md` still need to be updated.
- [ ] `git_basics_demo.md` needs further splitting. At least "My favorite neat little Git trick" should be separate.
- [ ] `containers_demo.md` needs further splitting.
- [ ] `homework_slides.md` -> `containers_homework_slides.md`
- [ ] `vm_containers_slides.md` -> `vm_containers_exercise`.md in chapter README
- [ ] `packaging_debian_slides.md` and similar: remove "packaging"
- [ ] hpc = spack

**Comment by IshaanDesai** [comment]
The `check-links` GitHub Action is expected to fail as the links are with reference to the `main` branch.

File: 00_organization/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+- What is research software engineering and what is special about it?
+- Why do we need research software engineering?
+
+| Duration | Content |
+| --- | --- |
+| 20 minutes | [`challenge_intro.md`](https://github.com/Simulation-Software-Engineering/Lecture-Material/blob/main/00_organization/challenge_intro_slides.md) |
+| 40 minutes | [`intro_course.md`](https://github.com/Simulation-Software-Engineering/Lecture-Material/blob/main/00_organization//intro_course_slides.md) |
+| 10 minutes | [`organizational_remarks_week3_slides.md`](https://github.com/Simulation-Software-Engineering/Lecture-Material/blob/main/00_organization/organizational_remarks_week3_slides.md) |
+| 5 minutes | [`organizational_remarks_week7_slides.md`](https://github.com/Simulation-Software-Engineering/Lecture-Material/blob/main/00_organization/organizational_remarks_week7_slides.md) |
+| 15 minutes | [`rse_basics.md`](https://github.com/Simulation-Software-Engineering/Lecture-Material/blob/main/organization/material/rse_basics_slides.md) |
```
Comment (ajaust) [inline 00_organization/README.md:15]: Should this chapter ("RSE Basics") be in the organization chapter or somewhere else? Is it organization or rather actual content?

File: 03_building_and_packaging/examples/cmake/main.cpp [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 int main(int argc, char *argv[])
 {
+  if (argc > 1)
+  {
+   
+  }
```
Comment (ajaust) [inline 03_building_and_packaging/examples/cmake/main.cpp:8]: Where does this change come from? Like... why is the content of this file changed?

File: scripts/create-pdf-from-markdown.sh [hunk 3]
```diff
@@ -12,13 +12,13 @@ else
     exit 1
 fi
```
Comment (ajaust) [inline scripts/create-pdf-from-markdown.sh:14]: Above (in line 11) replace

```bash
echo " Valid arguments are: slides, quiz, text, notes, or exercise"
```
by
```bash
echo " Valid arguments are: slides, quiz, demo, or exercise"
```

**Review by ajaust — CHANGES_REQUESTED** [review]
Looks very good already. I have the following questions, remarks, requests

- Please rename the `cpack` `debian` files such that they have consistent names. I also barely talk about Debian so I would suggest to remove "Debian" from the file names.
  - `cpack_debian_exercise.md`  -> `cpack_exercise.md`
  - `debian_slides.md` -> `cpack_slides.md`
  - `debian_demo.md` -> `cpack_demo.md`
- The files `intro_course_slides.md` and `challenge_intro_slides.md` in the organization chapter should get a consistent name. Intro should be either at the first or second position. 
- Please rename `vm_containers_exercise.md` into `virtualmachines_containers_exercise.md`. This is the only place where "virtualmachines" is not written out. 
- Please remove the directory `03_building_and_packaging/examples/cmake/build` from the PR. These are autogenerated files from CMake. About 85% of the files in this repository seem to be in this directory. A lot of files to review that should not be here. :wink: 
- Why did you integrate a link checker here? I think @uekerman and me had talked about this, but concluded to not add a link check. I cannot remember why. In general, I support the link checker, but we should maybe also aim for consistency between the repositories. In the [homepage repository](https://github.com/Simulation-Software-Engineering/homepage) we use `awesome_bot` as link checker.



Sorry for all problems I create by not carefully choosing file/chapter names. :scream:

File: 00_organization/README.md (hunk 1 above)
Comment (IshaanDesai) [reply 00_organization/README.md:15]: @uekerman suggestions?

Comment (uekerman) [reply 00_organization/README.md:15]: I would keep it here. Not worth a separate chapter right now. Maybe next year when we have this "all-in-one intro example", we could do a "Introduction" chapter.

**Comment by uekerman** [comment]
> Why did you integrate a link checker here? I think @uekerman and me had talked about this, but concluded to not add a link check. I cannot remember why. In general, I support the link checker, but we should maybe also aim for consistency between the repositories. In the homepage repository we use awesome_bot as link checker.

@IshaanDesai and me discussed this yesterday. I did not remember that we concluded on this and why :see_no_evil: 
Such a checker is needed for this refactoring. Otherwise, we will surely miss links. But, yes, we should not use different solutions here and in the homepage repo. 
@IshaanDesai: Could you take the solution from there?

**Comment by IshaanDesai** [comment]
> @IshaanDesai and me discussed this yesterday. I did not remember that we concluded on this and why see_no_evil Such a checker is needed for this refactoring. Otherwise, we will surely miss links. But, yes, we should not use different solutions here and in the homepage repo. @IshaanDesai: Could you take the solution from there?

Upvoting the reasoning of having a link checker. This PR now has the same link checker as the homepage repo :+1:

**Comment by IshaanDesai** [comment]
Failing workflows are expected to pass after merging and will be resolved on the `main` branch.

**Pull request merged**